hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/spark/sql/hudi/dml/others/TestMergeIntoTable.scala

```diff
@@ -1688,4 +1688,75 @@ class TestMergeIntoTable extends HoodieSparkSqlTestBase with ScalaAssertionSuppo
       }
     }
   }
+
+  test("Test partition not at end") {
+    withSQLConf("hoodie.memory.merge.max.size" -> "1",
+      "hoodie.memory.compaction.max.size" -> "1",
+      "hoodie.insert.shuffle.parallelism" -> "1",
+      "hoodie.upsert.shuffle.parallelism" -> "1",
+      "hoodie.bulkinsert.shuffle.parallelism" -> "1") {
```

Comment:
can we write a test that actually spills.
guess this test will succeed even w/o your fix right? 
essentially, the test should fail w/o your fix and should succeed w/ this fix.

what incase someone reverts your changes down the line due to some refactoring.

Reply:
It fails without the fix.

The memory size is set to 1, so it will spill on the second record.